### PR TITLE
[DataGridPro] Add `isRowDraggable` prop

### PR DESCRIPTION
**Pull request by DanailH**
Part of https://github.com/mui/mui-x/issues/4754
Based on this https://github.com/mui/mui-x/issues/4754#issuecomment-1190101937

`isRowDraggable` allows for the user to define which rows are draggable and which aren't.

### CODE REVIEW

File: packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx [hunk 1]
```diff
@@ -54,15 +59,18 @@ export const useGridRowReorder = (
...
+    },
+    [rowReordering, isRowDraggable, sortModel, treeDepth],
+  );
 
   const handleDragStart = React.useCallback<GridEventListener<'rowDragStart'>>(
     (params, event) => {
       // Call the gridEditRowsStateSelector directly to avoid infnite loop
       const editRowsState = gridEditRowsStateSelector(apiRef.current.state);
-      if (isRowReorderDisabled || Object.keys(editRowsState).length !== 0) {
+      if (isRowReorderDisabled(params) || Object.keys(editRowsState).length !== 0) {
```
Comment (m4theushw) [inline packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx:73]: Do we need this check if `rowDragStart` is already only fired if the row is draggable?

File: packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx [hunk 2]
```diff
@@ -41,11 +42,12 @@ const GridRowReorderCell = (params: GridRenderCellParams) => {
...
   const isDraggable = React.useMemo(
     () =>
-      !!rootProps.rowReordering &&
+      !!rowReordering &&
       !sortModel.length &&
       treeDepth === 1 &&
-      Object.keys(editRowsState).length === 0,
-    [rootProps.rowReordering, sortModel, treeDepth, editRowsState],
+      Object.keys(editRowsState).length === 0 &&
+      isRowDraggable(apiRef.current.getRowParams(params.id)),
```
Comment (m4theushw) [inline packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx:49]: Based on the comments in #4754 I don't know what users expect. They may want to keep some rows always at the bottom, like pinned rows but not pinned, or just disable the icon. If we disable the icon, then `isRowDraggable` could be only a prop of `GridRowReorderCell`, not for the grid. I think we should also add a demo of this feature, to clarify https://github.com/mui/mui-x/issues/4754#issuecomment-1187580566

File: packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx (hunk 1 above)
Comment (DanailH) [reply packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx:73]: Probably not, but you once might try and reorder rows programatically?

File: packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx (hunk 2 above)
Comment (DanailH) [reply packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx:49]: Noted, I'll add a demo. We can also go with the approach of adding a prop on the `GridRowReorderCell`. I went with the current approach as it's the closest in relation to what we have for the columns.
Regarding the first point of having pinned rows - same as the columns. You can reorder around them but can't reorder the one that's disabled.

**Comment by thuatole** [comment]
Hi @DanailH , why did we close this PR ? Is there any another approach to define which rows are draggable and which aren't ? Thanks

**Comment by DanailH** [comment]
@thuatole the main issue with this approach is that even if you mark a row as a nondraggable you can still change its position by reordering rows around it. That was @m4theushw's main concern. It would be interesting to know your use case of use this feature and why the row pinning feature doesn't work in your case.

**Pull request closed without merging**